File: leetCode/226invert-binary-tree/Solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {

    // public TreeNode invertTree(TreeNode root) {
    //     if (root == null) {
    //         return root;
    //       }
    //       // 分别翻转左、右子树
    //       TreeNode left = invertTree(root.left);
    //       TreeNode right = invertTree(root.right);
    //       // 翻转自己
    //       root.left = right;
    //       root.right = left;
    //       return root;
    // }

    public TreeNode invertTree(TreeNode root) {
        invert(root);
        return root;
    }

    public void invert(TreeNode curNode){
        if(curNode == null)return;
        //后续遍历法
        invert(curNode.left);
        invert(curNode.right);
        // if(curNode.left == null && curNode.right == null)return;
        // if(curNode.left != null)invert(curNode.left);
        // if(curNode.right != null)invert(curNode.right);

        TreeNode temp = curNode.left;
        curNode.left = curNode.right;
        curNode.right = temp;
    }

    public class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;
        TreeNode(int x) { val = x; }
    }
}

File: leetCode/226invert-binary-tree/Solution.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {TreeNode}
 */
var invertTree = function(root) {
    //树的翻转：每个节点的左右节点都做一次翻转
    if(root== null || (root.left == null && root.right == null))return root;
    var { left, right } = root;
    root.left = invertTree(right);
    root.right = invertTree(left);
    return root;  
};